homeassistant/components/loqed/config_flow.py

```diff
@@ -0,0 +1,137 @@
+"""Config flow for loqed integration."""
+from __future__ import annotations
+
+import json
+import logging
+from typing import Any
+
+import aiohttp
+from loqedAPI import loqed
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components import webhook
+from homeassistant.components.zeroconf import ZeroconfServiceInfo
+from homeassistant.const import CONF_HOST, CONF_WEBHOOK_ID
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(
+    hass: HomeAssistant, data: dict[str, Any], zeroconf_host: str | None
+) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+
+    newdata = data
+    json_config = json.loads(data["config"])
+    newdata["ip"] = json_config["bridge_ip"]
+    newdata["host"] = json_config["bridge_mdns_hostname"]
+    newdata["bkey"] = json_config["bridge_key"]
+    newdata["key_id"] = int(json_config["lock_key_local_id"])
+    newdata["api_key"] = json_config["lock_key_key"]
+
+    if zeroconf_host is not None and zeroconf_host != newdata["host"]:
+        raise InvalidAuth(
+            f"Got config for {newdata['host']} while configuring {zeroconf_host} "
+        )
+
+    # 1. Checking loqed-connection
+    try:
+        session = async_get_clientsession(hass)
+
+        apiclient = loqed.APIClient(session, "http://" + newdata["ip"])
+        api = loqed.LoqedAPI(apiclient)
+        lock = await api.async_get_lock(
+            newdata["api_key"], newdata["bkey"], newdata["key_id"], newdata["host"]
+        )
+        newdata["id"] = lock.id
+        # checking getWebooks to check the bridgeKey
+        await lock.getWebhooks()
+    except (aiohttp.ClientError):
+        _LOGGER.error("HTTP Connection error to loqed lock")
+        raise CannotConnect from aiohttp.ClientError
+    return newdata
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for loqed."""
+
+    VERSION = 2
```

Comment:
Sounds good. I bit of a hassle for beta users, but that's why it's beta ;)

Reply:
The beta integration is not a consideration for us. Additionally, I  hope your beta version is not using a conflicting domain, as domains need to be unique.

Also, changing this version number doesn't fix anything for a user on version 1 right now, as there is no migration? 🤷 

If that is not the case, I would strongly recommend adjusting the beta integration to have a different domain. E.g., `loqed_beta`.

Reply:
We'll cross that bridge when we get there. For now we'll stick with config flow V1

Reply:
I would not wait to cross that bridge. It will lead to unexpected errors and issue reports against our project. I would appreciate it if that is correct first. Thanks 👍

Reply:
That's why we're going with v1 and tell the (handful of) beta users to uninstall the unofficial integration first. When we see need for a full fledged beta we create a version with the loqed_beta domain